Clojure/tests/syntax_test_clojure.clj

```diff
@@ -0,0 +1,2007 @@
+;                ^ punctuation.section.parens.end.clojure
+  ([10/20 30/40])
+; ^ punctuation.section.parens.begin.clojure
+;  ^ punctuation.section.brackets.begin.clojure
+;   ^^^^^ constant.numeric.clojure
+;         ^^^^^ constant.numeric.clojure
+;              ^ punctuation.section.brackets.end.clojure
+;               ^ punctuation.section.parens.end.clojure
+  ([100.100 200.200])
+; ^ punctuation.section.parens.begin.clojure
+;  ^ punctuation.section.brackets.begin.clojure
+;   ^^^^^^^ constant.numeric.clojure
+;           ^^^^^^^ constant.numeric.clojure
+;                  ^ punctuation.section.brackets.end.clojure
+;                   ^ punctuation.section.parens.end.clojure
+  ([1e+10 2e-20])
+; ^ punctuation.section.parens.begin.clojure
+;  ^ punctuation.section.brackets.begin.clojure
+;   ^^^^^ constant.numeric.clojure
+;         ^^^^^ constant.numeric.clojure
+;              ^ punctuation.section.brackets.end.clojure
+;               ^ punctuation.section.parens.end.clojure
+
+; ## Invalid numbers
+
+  01234 +01234 -01234 '01234
+; ^^^^^ invalid.illegal.clojure
+;      ^- invalid
+;       ^^^^^^ invalid.illegal.clojure
+;              ^^^^^^ invalid.illegal.clojure
+;                     ^ constant.other.symbol.reader-macro.clojure
+;                      ^^^^^ invalid.illegal.clojure
+  01234N +01234N -01234N '01234N
+; ^^^^^^ invalid.illegal.clojure
+;       ^- invalid
+;        ^^^^^^^ invalid.illegal.clojure
+;                ^^^^^^^ invalid.illegal.clojure
+;                        ^ constant.other.symbol.reader-macro.clojure
+  10-20 10+20 1234n 1234m 1234. 1234.M
+; ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^- constant
+  10.0/20 10/20.0 10/+20 10/-20
+; ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^- constant
+  10:20:30
+; ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^- constant
+  1r000
+; ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^- constant
+
+; ## Ignore
+
+  ; valid symbols
+  .1234 .1234M
+; ^^^^^^^^^^^^- constant
+
+
+
+; # Symbols
+
+  ! $ % & * - _ = + | < > . / ?
+  ++ --
+  blah
+  blah/blah
+  blah.blah
+  blah.blah/blah
+  blah.blah/blah.blah
+  blah/blah/blah
+  blah1000
+  blah1000.blah1000
+  *blah*
+  blah'blah'
+  blah'''blah'''
+  blah:blah:blah
+  blah#blah#
+  blah///blah
+
+; ## Breaks
+
+  blah,blah,blah
+;     ^ comment.punctuation.comma.clojure
+  blah;blah;blah
+;     ^ comment.line.clojure punctuation.definition.comment
+  blah`blah
+;     ^ constant.other.symbol.reader-macro.clojure
+  blah~blah
+;     ^ constant.other.symbol.reader-macro.clojure
+  blah@blah
+;     ^ constant.other.symbol.reader-macro.clojure
+  blah^blah
+;     ^ constant.other.symbol.reader-macro.clojure
+  blah\blah
+;     ^^^^^ constant.character.clojure
+
+; ## Unaffected
+
+  'blah 'blah:blah
+; ^ constant.other.symbol.reader-macro.clojure
+;  ^- constant.other.symbol.reader-macro.clojure
+  [blah blah blah]
+
+; ## Invalid
+
+  //
+  blah:
+  blah::blah
+  /blah
+  blah/
+
+
+
+; # Keywords
+
+  :! :$ :% :& :* :- :_ := :+ :| :< :> :. :/ :?
```

Comment:
Are these keywords, or more like operators?

Reply:
There are actual valid keywords. On a side note, Clojure doesn't have any operators; things like `+` and `=` are regular identifiers.

Reply:
There is a `keyword` scope. Why did you not choose a descendant of that?

Reply:
So it would see to me from your description that the semantics of `:+` and `:blah` are the same, right? They should either be scoped the same, or if by convention they are used as "operators" we could scope them as such. I don't really see a reason they should be keywords, however.

Reply:
Currently they both have the `constant.language.keyword.clojure` scope.

I don't see any tests against a punctuation scope here, so scoping the colon as `punctuation.definition.keyword.clojure` sounds like a good addition.

Alternatively, if these are more similar to Ruby's symbol, maybe draw some inspiration for scope names from that?

Reply:
AFAICT, Ruby symbols are exactly like Clojure keywords. Looks like we all agree to follow the example and use `constant.other.keyword.clojure` for the entire constant and `punctuation.definition.keyword.clojure` for the colon.